graph/src/data/subgraph/mod.rs

```diff
@@ -837,12 +862,13 @@ impl UnvalidatedSubgraphManifest {
     /// Entry point for resolving a subgraph definition.
     /// Right now the only supported links are of the form:
     /// `/ipfs/QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k`
-    pub fn resolve(
+    pub async fn resolve(
         link: Link,
         resolver: Arc<impl LinkResolver>,
-        logger: Logger,
-    ) -> impl Future<Item = Self, Error = SubgraphManifestResolveError> + Send {
-        SubgraphManifest::resolve(link, resolver, logger).map(|manifest| Self(manifest))
+        logger: &Logger,
+    ) -> Result<Self, SubgraphManifestResolveError> {
+        let manifest = SubgraphManifest::resolve(link, resolver.deref(), logger).await?;
+        Ok(Self(manifest))
```

Comment:
Similarly here, this could also be
```rust
Ok(Self(SubgraphManifest::resolve(link, resolver.deref(), logger).await?))
```

Reply:
Just wanted to point out two examples, there's probably more places like this. Also note this is me being nit-picky, this is not something that I _expect_ to be changed. Up to you what you prefer 😄